sdk/core/azure-core/inc/azure/core/internal/http/user_agent_helper.hpp

```diff
@@ -12,13 +12,13 @@
 
 namespace Azure { namespace Core { namespace Http { namespace _internal {
   /**
-   * @brief Telemetry User-Agent string generator.
+   * @brief Telemetry User-Agent string helper.
    *
    */
-  class UserAgentGenerator {
+  class UserAgentHelper {
```

Comment:
Why did we decide to rename this class?

"Catch-all" terms like helper, utils, manager, etc. are a bit generic, so not sure if this is an improvement.

For context, there are a few common "catch-all" terms that people end up using, typically when there isn't a good name, or the abstraction/design doesn't have a clear purpose: https://stackoverflow.com/a/2071812

**If** we are going starting to go down the "helpers" route, we should re-evaluate what the intent of this class is here (and how we expect it to evolve), to help with naming.

Reply:
Rick felt that `*Generator` was not idiomatic C++., and was concerned that this would lead to a proliferation of Generator classes - I agree with him.

And we have a large set of existing `*Helper` classes so using the "Helper" suffix makes the most sense.

And the purpose of this class is simple: There are at least 3 different places within the Azure SDK that need to generate user agent strings (OpenTelemetry, Http Pipeline, Eventhubs, and eventually Servicebus and possibly Eventgrids). 

The logic to calculate the user agent is complicated and platform specific and we have always had the rule that if the same task needs to be done in more than one package, it should be moved to Azure Core.

Reply:
My preference here would be to call it `UserAgent`. Larry mentioned that's a little strange since the class is just a static method to construct the string. It doesn't actually encapsulate the value.

We don't have guidelines on the pattern for catch-all named classes. We should have one so that we are consistent across the SDKs.

Reply:
I totally understand why we need to have a type like this in Core (mainly the method), and for it to be _internal.

> And we have a large set of existing `*Helper` classes so using the "Helper" suffix makes the most sense.

I did not see any use of helper within the SDK **public** surface area (public or internal namespaces). I only them in private headers/_detail, tests, and sources (and a couple in samples, like GetEnvHelper).

Reply:
We are now using the existing `HttpShared` class as home for this static method, keeping naming the same.